model/task/task.go

```diff
@@ -2486,82 +2487,89 @@ func (t *Task) Insert() error {
 // are also archived.
 func (t *Task) Archive() error {
 	if t.DisplayOnly && len(t.ExecutionTasks) > 0 {
-		execTasks, err := FindAll(db.Query(ByIds(t.ExecutionTasks)))
+		err := ArchiveMany([]Task{*t})
+		return errors.Wrapf(err, "archiving display task '%s'", t.Id)
+	} else {
+		// Archiving a single task.
+		archiveTask := t.makeArchivedTask()
+		err := db.Insert(OldCollection, archiveTask)
 		if err != nil {
-			return errors.Wrap(err, "retrieving execution tasks")
+			grip.Error(message.WrapError(err, message.Fields{
+				"archive_task_id": archiveTask.Id,
+				"old_task_id":     archiveTask.OldTaskId,
+				"execution":       t.Execution,
+				"display_only":    t.DisplayOnly,
+			}))
+			return errors.Wrap(err, "inserting archived task into old tasks")
 		}
-		if err = ArchiveMany(execTasks); err != nil {
-			return errors.Wrap(err, "archiving execution tasks")
+		err = UpdateOne(
+			bson.M{IdKey: t.Id},
```

Comment:
Since we aren't using $inc anymore, should we include ExecutionKey in the update to ensure that we're updating what we think we are?

Reply:
Because tasks and display tasks are always highest execution, this can go back to using $inc!